Source/WebCore/platform/graphics/ca/GraphicsLayerCA.cpp

```diff
@@ -828,6 +829,41 @@ void GraphicsLayerCA::setOpacity(float opacity)
     noteLayerPropertyChanged(OpacityChanged);
 }
 
+void GraphicsLayerCA::setRenderingSuppressed(bool suppressed)
+{
+    GraphicsLayer::setRenderingSuppressed(suppressed);
+    recursiveSetAncestorHasRenderingSuppression(suppressed);
```

Comment:
It's weird for state to propagate to ancestors. Can't we just pick one of the root layers to set this on, and short-circuit the commits there?

Reply:
It's propagating to descendants. It's just so that we can short circuit painting for composited layers somewhere inside the suppressed subtree.

As you suggest below, we could short circuit `setNeedsDisplay` instead of painting itself, but we'd still need a way to know if the current layer is within a suppressed subtree (and I am using the assumption that an ancestor walk every time wouldn't be workable).